devito/interfaces.py

```diff
@@ -1,9 +1,47 @@
 import numpy as np
 from sympy import IndexedBase
 from tools import aligned
+from hashlib import sha1
+from random import randint
+import os
 
 
 class DenseData(IndexedBase):
+
+    # hash function used for generating part of memmap file name
+    _hashing_function = sha1
+    # this is the directory where memmap files will be created
+    _disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+
+    ## functions for managing memmap files
```

Comment:
Can we please keep all feature/functionality functions below `__init__()`? To my mind `__init__`/`__new__` should always be the first functions I see of any class.